packages/providers/src/provider.ts

```diff
@@ -618,6 +619,23 @@ export default class Provider {
     }));
   }
 
+  /**
+   * Returns Message Proof for given transaction id and the message id from MessageOut receipt
+   */
+  async getMessageProof(
+    /** The transaction to get message from */
+    transactionId: string,
+    /** The message id from MessageOut receipt */
+    messageId: string
+  ): Promise<GqlMessageProofFragmentFragment | null> {
```

Comment:
Do we want to wrap the Gql object so that it's not exposed? Similar to how the provider exports it's own defined `ChainInfo` object when querying chain info